app/CommonOptions.hs

```diff
@@ -82,6 +82,19 @@ parseGenericOutputFile = do
       )
   pure AppPath {_pathIsInput = False, ..}
 
+parseBuildDir :: Mod OptionFields (SomeBase Dir) -> Parser (AppPath Dir)
```

Comment:
I don't see that this belongs in CommonOptions. This is a concrete option. Imo this should be in app/GlobalOptions.hs